packages/remix-react/routes.tsx

```diff
@@ -175,10 +175,15 @@ function createLoader(route: EntryRoute, routeModules: RouteModules) {
   return loader;
 }
 
-function createAction(route: EntryRoute) {
-  if (!route.hasAction) return undefined;
-
+function createAction(route: EntryRoute, routeModules: RouteModules) {
   let action: ClientRoute["action"] = async ({ url, signal, submission }) => {
+    if (!route.hasAction) {
+      console.error(
+        `Route "${route.id}" does not have an action, but you are trying ` +
+          `to submit to it. To fix this, please add an \`action\` function to the route`
+      );
+    }
+
```

Comment:
Do we have a way to do this in dev only? I don't want bots POSTing all over the place making a mess of our logs. I think I'd rather just not even have this I think if we don't have a way to branch in dev vs. prod.

Reply:
We could also surface this message in the catch boundary data, but it would be the first case where the framework lands you there with data of its own. Would also mean you'd have to handle a data case that you did not throw yourself and I'm not a fan of having to have that knowledge.